src/core/support/cpu_linux.c

```diff
@@ -48,10 +48,10 @@
 #include <grpc/support/log.h>
 #include <grpc/support/sync.h>
 
-static int ncpus = 0;
+static unsigned ncpus = 0;
 
 static void init_num_cpus() {
-  ncpus = sysconf(_SC_NPROCESSORS_ONLN);
+  ncpus = (unsigned)sysconf(_SC_NPROCESSORS_ONLN);
```

Comment:
Unfortunately, I think this is wrong. `sysconf` apparently returns `-1` when a value cannot be determined.